_test/general.test.php

```diff
@@ -35,27 +90,322 @@ public function test_plugininfo() {
+
+        $this->assertEquals($expectedAttributeKey, $attributeKey, 'Expected attribute key in XML should match the category value when namespace height is two "cat".');
+        $this->assertEquals($expectedAttributeValue, $attributeValue, 'Expected attribute value in XML should match the namespace formatted in a FINDOLOGIC proper format when namespace height is two "test123_test123".');
+        $this->assertEquals($expectedOrdernumber, $ordernumber, 'Expected ordernumber in XML should match the pages namespace when namespace height is two "test123:test123".');
+    }
+
+    /**
+     * Test to ensure that ordernumber and attribute categories in the XML are valid when DokuWiki has a namespace height of one.
+     */
+    public function test_xml_elements_equal_to_dokuwiki_page_data_when_namespace_height_is_one()
+    {
+        $this->savePages(array('test321'));
+
+        $xml = $this->getXML();
+
+        $ordernumbers = $xml->xpath('/findologic/items/item/allOrdernumbers/ordernumbers/ordernumber');
+        $ordernumber = $ordernumbers[0];
+
+        $attributes = $xml->xpath('/findologic/items/item/allAttributes/attributes/attribute');
+        $attributeKey = (string)$attributes[0]->key[0];
+        $attributeValue = (string)$attributes[0]->values[0]->value[0];
+
+        $expectedOrdernumber = 'test321';
+        $expectedAttributeKey = 'cat';
+        $expectedAttributeValue = 'test321';
+
+        $this->assertEquals($expectedAttributeKey, $attributeKey, 'Expected attribute key in XML should match the category value when namespace height is two "cat".');
+        $this->assertEquals($expectedAttributeValue, $attributeValue, 'Expected attribute value in XML should match the namespace formatted in a FINDOLOGIC proper format when namespace height is two "test123_test123".');
+        $this->assertEquals($expectedOrdernumber, $ordernumber, 'Expected ordernumber in XML should match the pages namespace when namespace height is two "test123:test123".');
+    }
+
+    //======================================================================
+    // EXPORT CALL TESTS
+    //======================================================================
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     *
+     * @param integer $start Start value for Export call
+     * @param integer $count Count value for Export call
+     * @dataProvider parameterProviderForXMLCall
+     * @expectedException InvalidArgumentException
+     */
+    public function test_exception_is_thrown_when_call_export_with_start_value_one($start, $count)
+    {
+        $this->savePages(array('demopage1'));
+
+        $xml = $this->getXML($start, $count);
+        $this->fail('An exception should be thrown when trying to call the export with corner case start and count values.');
+    }
+
+    public function parameterProviderForXMLCall()
+    {
+        return [
+            'test start = 1 and count = 1' => [1, 1],
```

Comment:
A nice trick is to provide the assertion message as the third array element so you can fail the test with a well defined error message specific to this set of values.